### PR TITLE
MBL-1891, CHECK-356: Remove reference to Reward.shippingRules from estimated shipping

### PR DESCRIPTION
**Pull request by amy-at-kickstarter**
# 📲 What

Fetch shipping rules for a backing's reward on the Manage Pledge page.

This uses `SimpleShippingRulesExpanded` for the `Reward` attached to a `Backing`, and updates some plumbing to pass that fragment into the right place.

# 🤔 Why

This removes the only place in the app where we reference `Reward.shippingRules` directly. These are the shipping rules as set _by the creator_, instead of the shipping rules denormalized by country as displayed _to the backer_.

By removing this one reference, I can go ahead and remove `Reward.shippingRules` completely in the follow-up PR #2920. This will substantially simplify our shipping code.

# ✅ Acceptance criteria

- [x] Pledge page correctly shows estimated shipping
- [x] Late pledge page correctly shows estimated shipping
- [x] Manage pledge page correctly shows estimated shipping
- [x] Reward cards correctly show estimated shipping
- [x] Validate test case in #2200

### CODE REVIEW

File: KsApi/Sources/KsApi/models/graphql/adapters/Backing+BackingFragment.swift [hunk 1]
```diff
@@ -9,7 +9,8 @@ extension Backing {
   static func backing(
     from backingFragment: GraphAPI.BackingFragment,
     addOns: [Reward]? = nil,
-    paymentIncrements: [PledgePaymentIncrement] = []
+    paymentIncrements: [PledgePaymentIncrement] = [],
+    rewardExpandedShippingRules: [ShippingRule]? = nil,
```
Comment (amy-at-kickstarter) [inline KsApi/Sources/KsApi/models/graphql/adapters/Backing+BackingFragment.swift:13]: This extra plumbing is here because I only wanted to add the shipping rules to the query, instead of adding them to `BackingFragment`. Fetching shipping rules tends to be slow, so we only want to do it when it's necessary. `BackingFragment` is used in a few other queries.

File: Library/Sources/Library/Library/SharedFunctions.swift [hunk 2]
```diff
@@ -847,14 +847,10 @@ private func estimatedMinMax(
...
   rewards.forEach { reward in
     guard reward.shipping.enabled,
-          let shippingRules = reward.shippingRulesExpanded ?? reward.shippingRules else { return }
+          let shippingRules = reward.shippingRulesExpanded else { return }
 
     var shipping: ShippingRule? = shippingRules.first(where: { $0.location.id == locationId })
 
-    if shipping == nil && reward.shipping.preference == .unrestricted {
-      shipping = shippingRules.first
-    }
```
Comment (amy-at-kickstarter) [inline Library/Sources/Library/Library/SharedFunctions.swift:856]: This code existed to handle estimated shipping text on the Manage Pledge page. The Manage Pledge page wasn't fetching the expanded shipping rules attached to the backing's reward. It _was_ fetching the raw shipping rules, however. 

By making sure we always fetch the backing's reward's expanded shipping rules, I can clean this up.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
Backing+BackingFragment.swift | 90 | Optional should be implicitly initialized without nil (implicit_optional_initialization)


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by scottkicks — APPROVED** [review]
LGTM

Thanks for splitting this out into a separate PR. We should make sure to call this out in our Special Regressions QA checklist with a few of those edge case tests

**Comment by amy-at-kickstarter** [comment]
> LGTM
> 
> Thanks for splitting this out into a separate PR. We should make sure to call this out in our Special Regressions QA checklist with a few of those edge case tests

Good point, will do. I'll also see if I can get the late pledge case tested on Staging, as well, before I merge this.

**Comment by amy-at-kickstarter** [comment]
Rebasing against `main` because I merged the pre-cursor PR, #2918. And then merging!
